middleware/imagePixelation.js

```diff
@@ -0,0 +1,81 @@
+const axios = require("axios");
+const { createCanvas, loadImage } = require("canvas");
+
+async function pixelateImageFromURL(imageURL, chosenDifficulty = 2) {
+  const response = await axios.get(imageURL, { responseType: "arraybuffer" });
+  const originalImage = Buffer.from(response.data, "binary");
+
+  const imageElement = await loadImage(originalImage);
+  const minImageDimmension =
+    imageElement.width < imageElement.height
+      ? imageElement.width
+      : imageElement.height;
+  const pixelSize =
+    Math.floor(minImageDimmension / (60 / chosenDifficulty)) + 1;
+  const canvas = createCanvas(imageElement.width, imageElement.height);
+  const context = canvas.getContext("2d");
+
+  // Set canvas size to match the pixelated dimensions
+  canvas.width = imageElement.width - (imageElement.width % pixelSize);
+  canvas.height = imageElement.height - (imageElement.height % pixelSize);
```

Comment:
There is something weird going on with image height. If you go through every pixelation option you will notice that the size of the image gets smaller along with the intensity of the pixelation. The result is that you will see the "play settings" card jumping around.

Reply:
I have noticed this as well but i really could not figure out a way to fix it while keeping the other functionality in tact (keeping the image responsive, not having artifacts appear during the pixelation)